src/js/media-captions-list.js

```diff
@@ -0,0 +1,130 @@
+import { defineCustomElement } from './utils/defineCustomElement.js';
+import {
+  Window as window,
+  Document as document,
+} from './utils/server-safe-globals.js';
+import {
+  MediaUIEvents,
+  MediaUIAttributes,
+  TextTrackKinds,
+} from './constants.js';
+import { parseTextTrackStr, splitTextTracksStr } from './utils/captions.js';
+import MediaChromeListbox from './media-chrome-listbox.js';
+
+const CC_PREFIX = '!cc';
+const isCC = (extendedTextTrackStr = '') =>
+  extendedTextTrackStr?.startsWith?.(CC_PREFIX);
+const withoutCC = (extendedTextTrackStr = '') => {
+  if (!isCC(extendedTextTrackStr)) return extendedTextTrackStr;
+  return extendedTextTrackStr.slice(CC_PREFIX.length + 1);
+};
+
+const withCC = (textTrackStr = '') => `${CC_PREFIX}:${textTrackStr}`;
+
+const DEFAULT_UNKNOWN_LABEL = 'UNKNOWN';
+const DEFAULT_CC_LABEL_PART = '(cc)';
+const DEFAULT_FORMATTER = (value) => {
+  const textTrackStr = withoutCC(value);
+  const baseLabel =
+    parseTextTrackStr(textTrackStr)?.label ?? DEFAULT_UNKNOWN_LABEL;
+  const suffix = isCC(value) ? ` ${DEFAULT_CC_LABEL_PART}` : '';
+  return `${baseLabel}${suffix}`;
+};
+const parseAsSubCCStr = (str, kind = TextTrackKinds.SUBTITLES) => {
+  if (!str) return [];
+  if (kind === TextTrackKinds.SUBTITLES) return splitTextTracksStr(str);
+  return splitTextTracksStr(str).map(withCC);
+};
+
+class MediaCaptionsList extends MediaChromeListbox {
+  static get observedAttributes() {
+    return [
+      ...super.observedAttributes,
+      MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+      MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
+    ];
+  }
+
+  constructor({ format = DEFAULT_FORMATTER } = {}) {
+    super({ format });
+  }
+
+  attributeChangedCallback(attrName, oldValue, newValue) {
+    if (
+      [
+        MediaUIAttributes.MEDIA_SUBTITLES_LIST,
+        MediaUIAttributes.MEDIA_CAPTIONS_LIST,
+      ].includes(attrName) &&
+      oldValue !== newValue
+    ) {
+      this.data = [
+        ...parseAsSubCCStr(
+          this.getAttribute(MediaUIAttributes.MEDIA_SUBTITLES_LIST)
+        ),
+        ...parseAsSubCCStr(
+          this.getAttribute(MediaUIAttributes.MEDIA_CAPTIONS_LIST),
+          TextTrackKinds.CAPTIONS
+        ),
+      ];
+    } else if (
+      [
+        MediaUIAttributes.MEDIA_SUBTITLES_SHOWING,
+        MediaUIAttributes.MEDIA_CAPTIONS_SHOWING,
+      ].includes(attrName) &&
+      oldValue !== newValue
+      && !this.stateUpdaterId
+    ) {
+      // Discuss this simple scheduling as a means of avoiding edge case updates as a result of 
+      // unsetting + resetting via distinct events + separation of captions vs. subtitles (CJP)
+      this.stateUpdaterId = setTimeout(() => {
```

Comment:
@heff This is likely very safe, since it's going to defer to whatever the _**attribute**_ is (not the value passed in via the `attributeChangedCallback`. That said, I agree that its use case here is both a code sniff and, in my opinion, a workaround for some decisions earlier on where we made a "left" instead of a "right." These are related less to the generic "list" functionality and more to how we're modeling state for captions & subtitles. For example, even if we keep all of our existing attributes and events, I believe this code would be much cleaner/clearer if we had:

1. just one attribute for the list of both subtitles **_and_** captions
2. just one attribute for the currently selected subtitle **_or_** caption
3. just one event (figuring out the right name may be difficult) that will both "show" a given subtitle _**or**_ caption _**and**_ disable any currently showing subtitles/captions.

All of the acrobatics in `media-captions-list` are a result of the data coming from multiple attributes and the event to disable + the event to enable happening sequentially and in the same call stack, each of which will change the showing state of either captions or subtitles, so consecutively may change both.

I wanted to keep our existing larger architecture in media chrome "as is" with this spike effort, but I actually think it may be valuable to revisit some of those decisions now, given what this implementation effort is exposing.